### PR TITLE
bugfix: don't list dbt tests when using indirect selection

### PR DESCRIPTION
**Pull request by johannkm**
https://github.com/dagster-io/dagster/pull/20568 got messed up so that we still always provided all the check names in the dbt selection string. So that pr fixed singular tests, but not the cli arg length issue.

Fix that, and add tests on the cli arguments.

### CODE REVIEW

**Comment by johannkm** [comment]
* **#20944** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20944?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @johannkm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py [hunk 1]
```diff
@@ -40,6 +40,12 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
     yield from dbt.cli(["build"], context=context).stream()
 
 
+def _get_select_args(dbt_cli_invocation) -> Set[str]:
+    args = list(dbt_cli_invocation.process.args)
+    assert args[-2] == "--select"
+    return set(args[-1].split())
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py:46]: super nit: I'm allergic to indexing directly into a list. Let's just deconstruct?

```suggestion
    *_, dbt_select_flag, dbt_select_args = list(dbt_cli_invocation.process.args)
    
    assert dbt_select_flag == "--select"
    
    return set(dbt_select_args.split())
```

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py [hunk 2]
```diff
@@ -271,11 +277,15 @@ def _materialize_dbt_assets(
...
     selection: Optional[AssetSelection],
+    expected_dbt_selection: Optional[Set[str]] = None,
 ) -> ExecuteInProcessResult:
     @dbt_assets(manifest=manifest, dagster_dbt_translator=dagster_dbt_translator_with_checks)
     def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
         for dbt_command in dbt_commands:
-            yield from dbt.cli(dbt_command, context=context).stream()
+            cli_invocation = dbt.cli(dbt_command, context=context)
+            if expected_dbt_selection is not None:
+                assert _get_select_args(cli_invocation) == expected_dbt_selection
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py:287]:
```suggestion
            assert not expected_dbt_selection or _get_select_args(cli_invocation) == expected_dbt_selection
```

**Review by rexledesma — APPROVED** [review]

**Comment by SeppBerkner** [comment]
I am not sure if this is the cause of my issue I am facing, but in my comment https://github.com/dagster-io/dagster/issues/16997#issuecomment-2469940968 I mentioned that in linux system I still get an error that my dbt selection string is too long, whereas on my mac it is not an issue. Was there any work done regarding this?

**Pull request merged**